<launch>
  <!-- The below machine file makes control PC the
  "default" machine to execute nodes in. So it is not required to
  explicitly mention machine tag in the nodes that are called below.-->

  <arg name="config_file" default="$(find events_shaping_controller_control)/cfg/all_config_approach_exp2.yaml" />
  <rosparam command="load" file="$(arg config_file)" />

  <include file="$(find events_shaping_controller_control)/launch/setup.machine" />

  <!-- "Low-level" controllers to command Franka Emika Panda-->
  <include file="$(find events_shaping_controller_control)/launch/controllers.launch" pass_all_args="True">
    <arg name="robot_ip" value="192.168.100.11" />
  </include>

  <node name="nodes_sync" pkg="events_shaping_controller_control" type="nodes_synchronization.py">
  </node>

  <include file="$(find dvs_renderer)/launch/davis_mono.launch"/>


  <node name="image_publisher_with_bounding_box" pkg="events_shaping_controller_control" type="publish_image_with_bounding_box.py" >
  </node>

  <node name="approach_commander" pkg="events_shaping_controller_control" type="command_approach_constraint_divergence.py">
  </node>

  <!-- <node name="divergence_optimizer" pkg="events_shaping_controller_control" type="divergence_optimization.py">
  </node> -->

</launch>
